GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -472,17 +474,72 @@ public TestAccessor(RepoObjectsTree repoObjectsTree)
 
         private void mnuBtnUpdateAllRemotes_Click(object sender, EventArgs e)
         {
+            LoadPuttyKey();
             FormRemoteProcess.ShowDialog((GitModuleForm)ParentForm, "remote update");
         }
 
         private void mnuBtnPruneAllRemotes_Click(object sender, EventArgs e)
         {
+            LoadPuttyKey();
             FormRemoteProcess.ShowDialog((GitModuleForm)ParentForm, "remote update --prune");
         }
 
         private void mnuBtnPrune_Click(object sender, EventArgs e)
         {
+            LoadPuttyKey();
```

Comment:
`LoadPuttyKey` is run before each `FormRemoteProcess.ShowDialog` thus indicating it is a concern better placed with `FormRemoteProcess`.
It also doesn't seem correct to make the left panel responsible for loading or otherwise managing SSH keys.

`FormRemoteProcess` already has some functionality that looks like it interacts with Plink, e.g.
https://github.com/gitextensions/gitextensions/blob/027eabec3be497f8d780cc68ece268c64a43a7d5/GitUI/FormRemoteProcess.cs#L145-L153

Reply:
I also think that we should not duplicate `LoadPuttyKey()` code and do a refactoring to prevent it....

Reply:
Actually, there is already a good working implementation in place for "Fetch" (i.e. the option that was already implemented before #6293):

![](https://user-images.githubusercontent.com/460196/53209112-afae9200-3639-11e9-9ae4-04c8bdbd658d.png)

"Fetch" is connected to:
https://github.com/gitextensions/gitextensions/blob/96090c8d0114397fb0bcfbbb342261d1da45e904/GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Remotes.cs#L349-L357

Which uses the pull dialog, including it's `LoadPuttyKey()`:
https://github.com/gitextensions/gitextensions/blob/96090c8d0114397fb0bcfbbb342261d1da45e904/GitUI/CommandsDialogs/FormPull.cs#L739

I propose to extend the same mechanism for all newly added options. I have to modify  @pmiossec's implementation, but I think it's worth it. No code duplication and all options to fetch/update/prune remotes from the left panel will work the same way.